core/src/mindustry/world/blocks/defense/turrets/Turret.java

```diff
@@ -304,7 +313,11 @@ protected void turnToTarget(float targetRot){
         }
 
         public boolean shouldTurn(){
-            return true;
+            if(chargeTurn){
+                return true;
+            }else{
+                return !charging;
+            }
```

Comment:
I don't know what Anuke prefer, but with ternary operator it looks cleaner.

```suggestion
            return chargeTurn ? true : !charging;
```